SpiceSharp/Components/Noise/NoiseShot.cs

```diff
@@ -17,31 +19,33 @@ public class NoiseShot : NoiseGenerator
         /// Constructor
         /// </summary>
         /// <param name="name">Name of the noise source</param>
-        public NoiseShot(string name, int a, int b) : base(name, a, b) { }
+        public NoiseShot(string name, int node1, int node2) : base(name, node1, node2) { }
 
         /// <summary>
         /// Set the parameters of the shot noise
         /// </summary>
-        /// <param name="values">Values</param>
-        public override void Set(params double[] values)
+        /// <param name="coefficients">Values</param>
+        public override void SetCoefficients(params double[] coefficients)
         {
-            Current = values[0];
+            if (coefficients == null)
+                throw new ArgumentNullException(nameof(coefficients));
+            Current = coefficients[0];
```

Comment:
@svenboulanger Why use only one coefficients ?

Reply:
Shot noise has a uniform noise power density, only dependent on the current. I used multiple coefficients as a generalization for custom noise sources, but I don't really think it will ever be used. Would you like me to remove it?

Reply:
@svenboulanger It's ok. Please don't remove it.